stdlib/asm/crypto/stark/verifier.masm

```diff
@@ -0,0 +1,1464 @@
+    # From locaddr.AUX_RAND_ELEM to locaddr.AUX_RAND_ELEM + 16 - 1 we have the aux randomness
+    locaddr.AUX_RAND_ELEM
+    movdn.4
+    push.1
+    movdn.4         
+    # => [Seed, 1, rand_cstr_coef_ptr, X, Y]
+
+    repeat.16
+        exec.random_coin::generate_next_extfelt
+        mem_loadw.SEED
+    end
+    # => [Seed, 16, ptr + 16, X, Y]
+
+
+    # Reseed with auxiliary trace commitment
+    swapw
+    adv_loadw
+    loc_storew.AUX_TRACE_COM
+    swapw               
+    #=> [Seed, aux_trace_commitment, y, y, Y]
+    exec.random_coin::reseed         
+    mem_storew.SEED
+    #=> [Seed_n, Y, Y, y, y, Y]
+
+    # TODO: make use of these
+    dropw dropw drop drop  
+    #=> [Y, Y]
+    
+    #==============================================================================================
+    #       III) Draw constraint composition coefficients
+    #==============================================================================================
+
+    # Draw constraint composition coefficients
+    # From `compos_coef_ptr = locaddr.COMPOSITION_COEF` to locaddr.COMPOSITION_COEF + 235 - 1 (inclusive) we have the constraint 
+    # composition random coefficients.
+    # They are ordered in increasing index order and when loaded into the stack ordered as
+    # (r1_1, r1_0, r0_1, r0_0)
+    mem_loadw.SEED
+    locaddr.COMPOSITION_COEF
+    movdn.4
+    push.1
+    movdn.4         
+    # => [Seed, 1, compos_coef_ptr, Y]
+
+    repeat.235
+        exec.random_coin::generate_next_extfelt_tuple
+        mem_loadw.SEED
+        # => [Seed, 2, ptr + 1, ..]
+    end
+    # => [Seed, y, y, Y]
+   
+    #==============================================================================================
+    #       IV) Reseed with commitment to constraint composition polynomial H evaluations over LDE
+    #          and generate the Out-of-Domain (OOD) challenge z 
+    #==============================================================================================
+
+    # Reseed with constraint composition polynomial commitment
+    swapw
+    adv_loadw
+    loc_storew.COMPOSITION_POLY_COM
+    swapw
+    exec.random_coin::reseed
+    mem_storew.SEED
+    #=> [Seed_n, X, Y, y, y]
+
+    # Draw OOD z
+    swapw
+    mem_loadw.ZERO_ZERO_ZERO_ONE
+    swapw.2
+    dropw
+    push.1.1.0.0
+    hperm
+    swapw
+    drop drop
+    #=> [z1, z0, A, C, y, y]
+
+    # TODO: Reseed with z
+
+    # Store OOD challenge z and z^8 as [z^8_1, z^8_0, z1, z0]
+    dup.1
+    dup.1
+    repeat.3
+        dup.1
+        dup.1
+        ext2mul
+    end
+    mem_storew.Z
+    # => [X, Y, Y, x, x, 0, 0]
+    
+    #==============================================================================================
+    #       V) Read the OOD frames for the main trace, auxiliary trace and the trace of evaluations
+    #           of H over the LDE domain.
+    #==============================================================================================
```

Comment:
If we refactor `ood_frame` module as described in the previous comment, this section could be simplified to something like:
```
exec.odd_frame::load_evaluation_frame
exec::random_coint::reseed
exec::random_coint::reseed   # after winterfell refactoring this will not be needed

exec.ood_frame::load_constraint_evaluations
exec::random_coint::reseed

exec.ood_frame::compute_Hz
```
We could, of course, combine them into a single procedure - but

Reply:
Done!